server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -212,7 +211,7 @@ final class DefaultSearchContext extends SearchContext {
             engineSearcher.getQueryCachingPolicy(),
             lowLevelCancellation,
             executor,
-            shouldReverseLeafReaderContexts()
+            this
```

Comment:
Sorry, kind of consequence of the suggested change (thanks for accepting it), you don't need to pass the SearchContext instance anymore

Reply:
But then how will we get ShardSearchRequest objet here.

Reply:
Oops, please disregard, I see you still need it later on